### PR TITLE
add default interpreter for interpretHealth

### LINKED ISSUE #2354: [Umbrella] interpreter enable `InterpretHealth` feature
**What would you like to be added**:
We are going to enable `InterpretHealth` feature to [Resource Interpreter Framework](https://karmada.io/docs/userguide/globalview/customizing-resource-interpreter). 

Iteration tasks:
- [x] Make changes to [Resource Interpreter Framework](https://github.com/karmada-io/karmada/tree/master/pkg/resourceinterpreter) to enable `InterpretHealth`. (#2329)
- [x] Make changes to [karmada webhook](https://github.com/karmada-io/karmada/blob/b1684b646622456dcd02fade751c0dca278b33b9/pkg/webhook/configuration/validating.go#L65) to remove the restriction. (#2329)
- [x] Make changes to `Resource Interpreter Framework` to add build-in/default interpreter for resources in [here](https://github.com/karmada-io/karmada/blob/b1684b646622456dcd02fade751c0dca278b33b9/pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go#L25-L32).
- [x] Make changes to E2E tests to cover this feature. (#2416)

**Why is this needed**:
The [Graceful Eviction](https://github.com/karmada-io/karmada/issues/2281) feature needs to perform eviction/cleanup work as per resource health status.

### PR DESCRIPTION
**Pull request by zhuwint**
/kind feature

**What this PR does / why we need it**:

add default resource interpreter for interpretHealth

**Which issue(s) this PR fixes**:
Part of #2354

**Special notes for your reviewer**:

The method for health check is referred to https://github.com/jessesuen/argo-cd/blob/37a58fb9f1a2d7e20fe8e37a93ffa2e66b0eb9c7/util/health/health.go 

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

File: pkg/resourceinterpreter/defaultinterpreter/healthy.go [hunk 1]
```diff
@@ -31,11 +38,97 @@ func interpretDeploymentHealth(object *unstructured.Unstructured) (bool, error)
...
 }
+
+func interpretReplicaSetHealth(object *unstructured.Unstructured) (bool, error) {
+	replicaSet := &appsv1.ReplicaSet{}
+	err := helper.ConvertToTypedObject(object, replicaSet)
+	if err != nil {
+		return false, err
+	}
+
+	if replicaSet.Generation == replicaSet.Status.ObservedGeneration {
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/defaultinterpreter/healthy.go:57]: We can use guard code here, just like:
```
	if replicaSet.Generation != replicaSet.Status.ObservedGeneration {
		return false, nil
	}
```
Same as bellow.

File: pkg/resourceinterpreter/defaultinterpreter/healthy.go [hunk 2]
```diff
@@ -31,11 +38,97 @@ func interpretDeploymentHealth(object *unstructured.Unstructured) (bool, error)
...
+
+func interpretPersistentVolumeClaimHealth(object *unstructured.Unstructured) (bool, error) {
+	pvc := &corev1.PersistentVolumeClaim{}
+	err := helper.ConvertToTypedObject(object, pvc)
+	if err != nil {
+		return false, err
+	}
+
+	healthy := pvc.Status.Phase == corev1.ClaimBound
+	return healthy, nil
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/defaultinterpreter/healthy.go:133]: How about this:
```go
return pvc.Status.Phase == corev1.ClaimBound, nil
```

File: pkg/resourceinterpreter/defaultinterpreter/healthy_test.go [hunk 3]
```diff
@@ -211,3 +212,314 @@ func Test_interpretStatefulSetHealth(t *testing.T) {
 		})
 	}
 }
+
+func Test_interpretReplicaSetHealth(t *testing.T) {
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/defaultinterpreter/healthy_test.go:216]: The case for determining the replicas is missing.

File: pkg/resourceinterpreter/defaultinterpreter/healthy_test.go [hunk 4]
```diff
@@ -211,3 +212,314 @@ func Test_interpretStatefulSetHealth(t *testing.T) {
...
+			name: "failed convert to ReplicaSet object",
+			object: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec":   "format error",
+					"status": "format error",
+				},
+			},
+			want:    false,
+			wantErr: true,
+		},
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/defaultinterpreter/healthy_test.go:233]: I see similar judgments in each test case. Is it possible to consider testing method `ConvertToTypedObject` so that similar test cases can be omitted?

If this is feasible, you can submit a separate pr to test method `ConvertToTypedObject`.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/lgtm
/approve

**Review by RainbowMango — APPROVED** [review]
/approve
